src/test/java/bio/terra/service/dataset/DatasetDaoTest.java

```diff
@@ -1214,4 +1215,14 @@ public void testRetrieveExclusivelyLockedDataset() throws Exception {
         lockedDatasetSummaryEnumerationItemExclusiveLock,
         equalTo(flightId));
   }
+
+  @Test
+  public void testRetrieveSharedLock() throws Exception {
+    UUID datasetId = createDataset("dataset-minimal.json");
+    String flightId = ShortUUID.get();
+    datasetDao.lockShared(datasetId, flightId);
+    Dataset lockedDataset = datasetDao.retrieve(datasetId);
+    var datasetSharedLocks = ResourceLocksUtils.getSharedLock(lockedDataset.getResourceLocks());
+    assertThat("Correct shared lock is returned", datasetSharedLocks.get(0), equalTo(flightId));
```

Comment:
I'd recommend this assertion instead, which also verifies that there is exactly one lock matching your expectation:

```
assertThat("Correct shared lock is returned", datasetSharedLocks, contains(flightId));
```